### PR TITLE
Closed many potentially open streams

### PR DESCRIPTION
**Pull request by daiLlew**
### What

Added try-with-resources blocks around many potentially open streams to try and fix file descriptors issue.

### How to review

Zebedee should work exactly the same after before.

### Who can review

@ian-kent @CarlHembrough @Crispioso @LloydGriffiths @jondewijones

### CODE REVIEW

File: zebedee-reader/src/main/java/com/github/onsdigital/zebedee/search/indexing/Indexer.java [hunk 1]
```diff
@@ -131,8 +132,10 @@ private void loadDepartments() throws IOException {
 
         elasticSearchLog("Indexing departments").log();
         long start = System.currentTimeMillis();
-        InputStream resourceStream = SearchBoostTermsResolver.class.getResourceAsStream("/search/departments/departments.txt");
-        try (BufferedReader br = new BufferedReader(new InputStreamReader(resourceStream))) {
+        try (
+                InputStream resourceStream = SearchBoostTermsResolver.class.getResourceAsStream(DEPARTMENTS_PATH);
+                BufferedReader br = new BufferedReader(new InputStreamReader(resourceStream))
+        ) {
```
Comment (CarlHembrough) [inline zebedee-reader/src/main/java/com/github/onsdigital/zebedee/search/indexing/Indexer.java]: Consider declaring the InputStreamReader in the try with resources so it is closed if the BufferedReader constructor throws an exception.

File: zebedee-reader/src/main/java/com/github/onsdigital/zebedee/search/indexing/SearchBoostTermsResolver.java [hunk 2]
```diff
@@ -33,8 +37,10 @@ public static void loadTerms() throws IOException {
 
 
     static void loadTerms(String fileName) throws IOException {
-        InputStream resourceStream = SearchBoostTermsResolver.class.getResourceAsStream(fileName);
-        try (BufferedReader br = new BufferedReader(new InputStreamReader(resourceStream))) {
+        try (
+                InputStream resourceStream = SearchBoostTermsResolver.class.getResourceAsStream(fileName);
+                BufferedReader br = new BufferedReader(new InputStreamReader(resourceStream))
+        ) {
```
Comment (CarlHembrough) [inline zebedee-reader/src/main/java/com/github/onsdigital/zebedee/search/indexing/SearchBoostTermsResolver.java]: Consider declaring the InputStreamReader in the try with resources so it is closed if the BufferedReader constructor throws an exception.

File: zebedee-cms/src/main/java/com/github/onsdigital/zebedee/api/Table.java [hunk 3]
```diff
@@ -68,24 +68,26 @@ public void createTable(HttpServletRequest request, HttpServletResponse response
...
-        Audit.Event.COLLECTION_TABLE_CREATED
-                .parameters()
-                .host(request)
-                .collection(collection)
-                .user(session.email)
-                .log();
+        try (Resource resource = collectionReader.getResource(uri)) {
+
+            TableModifications modifications = getTableModifications(request);
+            Node table = XlsToHtmlConverter.convertToHtmlPageWithModifications(resource.getData(), modifications);
```
Comment (daiLlew) [inline zebedee-cms/src/main/java/com/github/onsdigital/zebedee/api/Table.java]: I did see that - I did a google search to try and work out what should be closed but couldn't find an answer. Their API doc doesn't say anything either.

Comment (daiLlew) [inline zebedee-cms/src/main/java/com/github/onsdigital/zebedee/api/Table.java]: will do.

File: zebedee-cms/src/main/java/com/github/onsdigital/zebedee/service/BabbagePdfService.java [hunk 4]
```diff
@@ -48,19 +49,20 @@ public InputStream generatePdf(String uri) throws IOException {
...
+            HttpGet httpGet = new HttpGet(src);
+            httpGet.addHeader("Cookie", "access_token=" + session.id);
+            httpGet.addHeader("Cookie", "collection=" + collection.description.id);
 
-        if (response.getStatusLine().getStatusCode() != 200) {
-            throw new IOException("Failed to generate PDF for URI " + uri +
-                    ". Response: " + response.getStatusLine().getStatusCode() +
-                    " " + response.toString());
-        }
+            try (CloseableHttpResponse response = client.execute(httpGet)) {
```
Comment (daiLlew) [inline zebedee-cms/src/main/java/com/github/onsdigital/zebedee/service/BabbagePdfService.java]: I did check this locally before pushing this change and I didn't get this issue. I will check again anyway.

**Pull request merged**